src/fsharp/TastOps.fs

```diff
@@ -3004,7 +3004,22 @@ let isByrefLikeTy g m ty =
 
 let isSpanLikeTy g m ty =
     isByrefLikeTy g m ty && 
-    not (isByrefTy g ty) 
+    not (isByrefTy g ty)
+
+let isSpanTyconRef (g: TcGlobals) m (tcref: TyconRef) =
+    if isByrefLikeTyconRef g m tcref then
+        match tcref.ILTyconInfo with
+        | TILObjectReprData(_, _, ilTypeDef) -> ilTypeDef.Name = "System.Span`1"
+    else
+        false
+    
+let isSpanTy g m ty =
+    ty |> stripTyEqns g |> (function TType_app(tcref, _) -> isSpanTyconRef g m tcref | _ -> false)
```

Comment:
Please use a simple match

Reply:
Not sure what you mean, but we do this kind of thing in TastOps.fs already. Can you show me exactly what code you would like instead?

Reply:
I mean please don't use function keyword. In most cases code turns out to be more readable without it

Reply:
This topic on `function` is pretty subjective. For me, I find it readable and that's just my own opinion. However, we do this a lot in TastOps.fs and I'm following the current pattern that is there. It will take a bit more convincing for me to change this otherwise.

Reply:
It's just a suggestion - and of course compiler code is different than other code. Not really important here.

Reply:
@TIHan is right, he's just following the coding that's there